adr/0004-supported-python-versions.md

```diff
@@ -0,0 +1,36 @@
+# Supported Python Versions
+
+Date: 2017-11-20
+
+## Status
+
+Proposed
+
+## Context 
+
+eLife has numerous projects written completely and partly with the [Python programming language](https://www.python.org/).
+
+In order to provide language version consistency across projects we need to get a consensus on which versions we are going to support. 
+
+## Descision
+
+We will use Python 2.7.14 as our default version for any project that solely uses or supports Python 2.
+
+We will use Python 3.6.3 as our default supported version for any project that solely uses or supports Python 3.
```

Comment:
To make things simpler we could require projects that use 3.x (hence 3.6 according to this) to first upgrade to Ubuntu 16.04.
However, 16.04 still provides only 3.5 by default so we should find a good PPA for 3.6.

Reply:
https://launchpad.net/~jonathonf/+archive/ubuntu/python-3.6 seems to be the go to.